File: src/services/CartService.js
import axios from "axios";

const API_BASE = "http://localhost:8080/cart";

const CartService = {
    getCart: (accountId) =>
        axios
            .get(`${API_BASE}/getCart/${accountId}`)
            .then((res) => res.data.data),

    removeItem: (accountId, skuCode) =>
        axios
            .put(`${API_BASE}/removeItem/${accountId}?skuCode=${encodeURIComponent(skuCode)}`)
            .then((res) => res.data.data),

    updateItem: (cartRequest) =>
        axios
            .post(`${API_BASE}/updateItem`, cartRequest)
            .then((res) => res.data.data),

    clearCart: (accountId) =>
        axios.post(`${API_BASE}/clearCart/${accountId}`),
};

export default CartService;

File: project/src/main/java/com/example/clothing_store/service/Cart/CartService.java
package com.example.clothing_store.service.Cart;

import com.example.clothing_store.dto.CartDTO;
import com.example.clothing_store.dto.CartRequest;
import com.example.clothing_store.response.CommonResponse;
import com.example.clothing_store.response.ResponseToData;
import org.springframework.stereotype.Service;

@Service
public interface CartService {
    CartDTO getOrCreateCart(int accountId);

    CartDTO addItem(CartRequest cartRequest);

    ResponseToData<CartDTO> removeItem(int accountId, String skuCode);

    CartDTO updateItem(CartRequest cartRequest);

    CommonResponse clearCart(int accountId);
}
